src/lifecycle/properties-init.js

```diff
@@ -150,6 +161,18 @@ function createNativePropertyDefinition (name, opts) {
       newValue = opts.coerce(newValue);
     }
 
+    const propertyHasChanged = newValue !== oldValue;
+    if (propertyHasChanged && opts.event) {
+      const cancelled = triggerPropertyChangedEvent(this, {
```

Comment:
Is it worth having to call out to another function for this? I don't see it abstracting much.